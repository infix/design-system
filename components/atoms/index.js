import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
