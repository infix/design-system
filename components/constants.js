/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
