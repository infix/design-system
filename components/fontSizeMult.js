/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
