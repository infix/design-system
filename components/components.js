import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
