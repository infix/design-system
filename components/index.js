import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
