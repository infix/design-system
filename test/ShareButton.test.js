/* global describe, test, expect */
import React from 'react'
import { renderJSON } from './index'
import { TwitterShareButton, FacebookShareButton } from '../components'

describe('TwitterShareButton', () => {
  test('renders', () => {
    const json = renderJSON(
      <TwitterShareButton />
    )
    expect(json).toMatchSnapshot()
  })
})

describe('FacebookShareButton', () => {
  test('renders', () => {
    const json = renderJSON(
      <FacebookShareButton />
    )
    expect(json).toMatchSnapshot()
  })
})
