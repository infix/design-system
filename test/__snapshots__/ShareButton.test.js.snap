// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookShareButton renders 1`] = `
.c0 {
  color: blue;
}

.c3 {
  border-radius: 32px;
  display: inline-block;
  height: 40px;
  line-height: 1;
  padding-left: 32px;
  padding-right: 32px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c3:hover {
  border: none;
  background-color: #37a6ed;
}

.c3:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c2 {
  font-size: 16px;
  background-color: #3b5998;
}

.c1 {
  text-transform: none;
}

.c1:hover {
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
  background-color: #3b5998;
}

.c1:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
  background-color: #3b5998;
}

.c4 {
  padding-left: 8px;
}

<FacebookShareButton
  fontSize={2}
  url=""
>
  <ForwardRef
    color="blue"
    href="https://www.facebook.com/sharer/sharer.php?&u="
    is="a"
  >
    <ForwardRef
      className="c0"
      color="blue"
      href="https://www.facebook.com/sharer/sharer.php?&u="
      is="a"
    >
      <a
        className="c0"
        href="https://www.facebook.com/sharer/sharer.php?&u="
      >
        <ShareButton__StyledShareButton
          bg="#3b5998"
        >
          <Button
            bg="#3b5998"
            border={2}
            className="c1"
            fontSize={1}
            hollow={false}
            inverted={false}
          >
            <Button__StyledButton
              bg="#3b5998"
              border={2}
              className="c1 c2"
              fontSize={1}
              hollow={false}
              inverted={false}
            >
              <button
                className="c1 c2 c3"
                fontSize={1}
              >
                <FaFacebook>
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m29.4 0.3v5.9h-3.5q-1.9 0-2.6 0.8t-0.7 2.4v4.2h6.6l-0.9 6.6h-5.7v16.9h-6.8v-16.9h-5.7v-6.6h5.7v-4.9q0-4.1 2.3-6.4t6.2-2.3q3.3 0 5.1 0.3z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </FaFacebook>
                <ShareButton__StyledText
                  pl={2}
                >
                  <span
                    className="c4"
                  >
                    Share
                  </span>
                </ShareButton__StyledText>
              </button>
            </Button__StyledButton>
          </Button>
        </ShareButton__StyledShareButton>
      </a>
    </ForwardRef>
  </ForwardRef>
</FacebookShareButton>
`;

exports[`TwitterShareButton renders 1`] = `
.c0 {
  color: blue;
}

.c3 {
  border-radius: 32px;
  display: inline-block;
  height: 40px;
  line-height: 1;
  padding-left: 32px;
  padding-right: 32px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c3:hover {
  border: none;
  background-color: #37a6ed;
}

.c3:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c2 {
  font-size: 16px;
  background-color: #1DA1F2;
}

.c1 {
  text-transform: none;
}

.c1:hover {
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
  background-color: #1DA1F2;
}

.c1:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
  background-color: #1DA1F2;
}

.c4 {
  padding-left: 8px;
}

<TwitterShareButton
  fontSize={2}
  hashtags=""
  inReplyTo=""
  message=""
  url=""
  via=""
>
  <ForwardRef
    color="blue"
    href="https://twitter.com/intent/tweet?text="
    is="a"
  >
    <ForwardRef
      className="c0"
      color="blue"
      href="https://twitter.com/intent/tweet?text="
      is="a"
    >
      <a
        className="c0"
        href="https://twitter.com/intent/tweet?text="
      >
        <ShareButton__StyledShareButton
          bg="#1DA1F2"
        >
          <Button
            bg="#1DA1F2"
            border={2}
            className="c1"
            fontSize={1}
            hollow={false}
            inverted={false}
          >
            <Button__StyledButton
              bg="#1DA1F2"
              border={2}
              className="c1 c2"
              fontSize={1}
              hollow={false}
              inverted={false}
            >
              <button
                className="c1 c2 c3"
                fontSize={1}
              >
                <FaTwitter>
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </FaTwitter>
                <ShareButton__StyledText
                  pl={2}
                >
                  <span
                    className="c4"
                  >
                    Tweet
                  </span>
                </ShareButton__StyledText>
              </button>
            </Button__StyledButton>
          </Button>
        </ShareButton__StyledShareButton>
      </a>
    </ForwardRef>
  </ForwardRef>
</TwitterShareButton>
`;
